test/framework/machine_helpers.go

```diff
@@ -21,32 +21,66 @@ import (
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
-	"github.com/pkg/errors"
 
+	"github.com/pkg/errors"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+type GetMachinesByClusterInput struct {
+	Lister      Lister
+	ClusterName string
+	Namespace   string
+}
+
+// GetMachinesByCluster returns the Machine objects that matches provided options for a cluster.
+// Important! this method relies on labels that are created by the CAPI controllers during the first reconciliation, so
+// it is necessary to ensure this is already happened before calling it.
+func GetMachinesByCluster(ctx context.Context, input GetMachinesByClusterInput, options ...client.ListOption) []clusterv1.Machine {
+	opts := append(byClusterOptions(input.ClusterName, input.Namespace), options...)
+
+	machineList := &clusterv1.MachineList{}
+	Expect(input.Lister.List(ctx, machineList, opts...)).To(Succeed(), "Failed to list MachineList object for Cluster %s/%s", input.Namespace, input.ClusterName)
+
+	return machineList.Items
+}
+
+// deploymentNameMachineOptions returns a set of ListOptions that allows to get all machine objects belonging to a machine deployment.
+func deploymentNameMachineOptions(deploymentName string) []client.ListOption {
+	return []client.ListOption{
+		client.MatchingLabels{clusterv1.MachineDeploymentLabelName: deploymentName},
+	}
+}
+
+// GetMachinesByClusterAndDeployment returns the Machine objects for a cluster belonging to a machine deployment.
+// Important! this method relies on labels that are created by the CAPI controllers during the first reconciliation, so
+// it is necessary to ensure this is already happened before calling it.
+func GetMachinesByClusterAndDeployment(ctx context.Context, deploymentName string, input GetMachinesByClusterInput) []clusterv1.Machine {
```

Comment:
I think we should add a specific input type to be consistent with all the other helpers.

Reply:
+1 to what @fabianofranz said here. You can likely even embed GetMachinesByClusterInput in the new input type